libs/cardano-ledger-core/src/Cardano/Ledger/CompactAddress.hs

```diff
@@ -590,12 +619,10 @@ instance CC.Crypto crypto => ToCBOR (CompactAddr crypto) where
   toCBOR (UnsafeCompactAddr bytes) = toCBOR bytes
 
 instance CC.Crypto crypto => FromCBOR (CompactAddr crypto) where
-  fromCBOR = snd <$> fromCborBackwardsBothAddr
-
--- sbs <- fromCBOR
--- case deserializeShortAddr @crypto sbs of
---   Just _ -> pure $ UnsafeCompactAddr sbs
---   Nothing -> cborError $ DecoderErrorCustom "Addr" "invalid address"
+  fromCBOR = do
+    (_addr, cAddr) <- fromCborBackwardsBothAddr
+    pure cAddr
```

Comment:
Why the change here?

Reply:
This was mainly an experiment to check if for some reason `fmap snd` is too lazy for the `Decoder` and forcing the tuple would make an improvement. Honestly I don't know if it did, because I was also adding INLINE pragmas as well. As soon as I saw a regression gone, I didn't bother going back and changing it back to `snd <$> fromCborBackwardsBothAddr`, since that would mean at least another hour being wasted. However, if you are really curious I can make an effort and do a db-analyser run with this change reverted.